src/engine/QueryPlanner.cpp

```diff
@@ -1819,6 +1824,28 @@ auto QueryPlanner::createJoinWithService(
   return plan;
 }
 
+// _____________________________________________________________________
+auto QueryPlanner::createOptionalJoinWithService(const SubtreePlan& a,
+                                                 const SubtreePlan& b)
+    -> std::optional<SubtreePlan> {
+  // We can only proceed if (only) the right subtree is a `SERVICE`.
+  auto aRootOp = std::dynamic_pointer_cast<Service>(a._qet->getRootOperation());
+  auto bRootOp = std::dynamic_pointer_cast<Service>(b._qet->getRootOperation());
```

Comment:
Why is it important that the left input is not a service? I think this completely doesn't matter?